stdlib/public/core/UnsafeUninitializedPointer.swift

```diff
@@ -0,0 +1,140 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// Provides scoped access to a raw buffer pointer with the specified byte count
+/// and alignment.
+///
+/// - Parameters:
+///   - byteCount: The number of bytes to temporarily allocate. `byteCount` must
+///     not be negative.
+///   - alignment: The alignment of the new, temporary region of allocated
+///     memory, in bytes.
```

Comment:
I'm not sure what the right phrasing is, but we should have wording for "compile-time constant" here and in other places where that is a requirement.

Reply:
Given that there is no requirement that the alignment be a compile-time constant, I don't believe such wording is needed.

Reply:
Maybe I'm misunderstanding something... There's this call in the body:

```
Builtin.stackAlloc(byteCount._builtinWordValue, alignment._builtinWordValue)
```

For this call to work, `alignment._builtinWordValue` needs to be a compile-time constant according the documentation for `stackAlloc` in `Builtins.def`. Doesn't that imply that `alignment` also needs to be a compile-time constant?

Reply:
I think it just implies my comment in Builtin.defs is leaking implementation details. :)